src/device/riscv/handleinterrupt64.S

```diff
@@ -0,0 +1,44 @@
+.section .text.handleInterruptASM
+.global handleInterruptASM
+.type handleInterruptASM,@function
+handleInterruptASM:
+    // Save and restore all registers, because the hardware only saves/restores
+    // the pc.
+    // Note: we have to do this in assembly because the "interrupt"="machine"
+    // attribute is broken in LLVM: https://bugs.llvm.org/show_bug.cgi?id=42984
```

Comment:
Sidenote (unrelated to this PR): this should be fixed with LLVM 11 so that we won't need this workaround anymore.